### PR TITLE
Add Exception instance for TOMLError

### LINKED ISSUE #13: Add Exception TOMLError instance
Implement `displayException` with `renderTOMLError`

### PR DESCRIPTION
**Pull request by tbidne**
Resolves #13.

Hi! First, thanks for the great library. The API is excellent.

I am happy to discover this issue, as I thought of the same thing when using it :slightly_smiling_face:.

### CODE REVIEW

File: src/TOML/Error.hs [hunk 1]
```diff
@@ -24,6 +26,11 @@ data TOMLError
   | DecodeError DecodeContext DecodeError
   deriving (Show, Eq)
 
+instance Exception TOMLError where
+#if MIN_VERSION_base(4,8,0)
```
Comment (brandonchinn178) [inline src/TOML/Error.hs]: What version of ghc does base 4.8 come with? i suspect the lowest version of ghc we test against is more recent than this

Comment (tbidne) [reply src/TOML/Error.hs]: According to the [wiki](https://gitlab.haskell.org/ghc/ghc/-/wikis/commentary/libraries/version-history), `7.8.4` is the latest version that is too low, so I am sure you are right. I am happy to remove the cpp if you like.

Comment (brandonchinn178) [reply src/TOML/Error.hs]: Yes please; whatever the oldest stack.yaml we have in the repo is the oldest version of deps we support

Comment (tbidne) [reply src/TOML/Error.hs]: Done.

**Comment by brandonchinn178** [comment]
Thanks!

**Comment by tbidne** [comment]
Thanks for merging!

**Pull request merged**